internal/service/advancedclustertpf/resource.go

```diff
@@ -99,80 +96,230 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
+	clusterName := plan.Name.ValueString()
+	var cluster *admin20240805.ClusterDescription20240805
+	var advConfig *admin20240805.ClusterDescriptionProcessArgs20240805
+	var legacyAdvConfig *admin20240530.ClusterDescriptionProcessArgs
+	var err error
 	if patchReq != nil {
-		err := StoreUpdatePayload(patchReq)
-		if err != nil {
-			diags.AddError("error storing update payload", fmt.Sprintf("error storing update payload: %s", err.Error()))
+		pauseAfter := false
+		if patchReq.Paused != nil && patchReq.GetPaused() {
+			// More changes than pause, need to pause after
+			if !reflect.DeepEqual(pauseRequest, *patchReq) {
+				pauseAfter = true
+				patchReq.Paused = nil
+			}
+		} else if patchReq.Paused != nil && !patchReq.GetPaused() {
+			// More changes than pause, need to resume before applying changes
+			if !reflect.DeepEqual(resumeRequest, *patchReq) {
+				patchReq.Paused = nil
+				cluster = r.updateAndWait(ctx, &resumeRequest, diags, &plan)
+				if cluster == nil {
+					return
+				}
+			}
+		}
+		cluster = r.updateAndWait(ctx, patchReq, diags, &plan)
+		if cluster == nil {
+			return
+		}
+		if pauseAfter {
+			cluster = r.updateAndWait(ctx, &pauseRequest, diags, &plan)
 		}
 	}
 	patchReqProcessArgs := update.PatchPayloadTpf(ctx, diags, &state.AdvancedConfiguration, &plan.AdvancedConfiguration, NewAtlasReqAdvancedConfiguration)
 	if patchReqProcessArgs != nil {
-		err := StoreUpdatePayloadProcessArgs(patchReqProcessArgs)
+		advConfig, _, err = api.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, patchReqProcessArgs).Execute()
 		if err != nil {
-			diags.AddError("error storing update payload advanced config", fmt.Sprintf("error storing update payload: %s", err.Error()))
+			diags.AddError("errorUpdateAdvancedConfig", fmt.Sprintf(errorConfigUpdate, clusterName, err.Error()))
 		}
 	}
 	patchReqProcessArgsLegacy := update.PatchPayloadTpf(ctx, diags, &state.AdvancedConfiguration, &plan.AdvancedConfiguration, NewAtlasReqAdvancedConfigurationLegacy)
 	if patchReqProcessArgsLegacy != nil {
-		err := StoreUpdatePayloadProcessArgsLegacy(patchReqProcessArgsLegacy)
+		legacyAdvConfig, _, err = r.Client.AtlasV220240530.ClustersApi.UpdateClusterAdvancedConfiguration(ctx, projectID, clusterName, patchReqProcessArgsLegacy).Execute()
 		if err != nil {
-			diags.AddError("error storing update payload advanced config legacy", fmt.Sprintf("error storing update payload: %s", err.Error()))
+			diags.AddError("errorUpdateAdvancedConfigLegacy", fmt.Sprintf(errorConfigUpdate, clusterName, err.Error()))
 		}
 	}
-	if diags.HasError() {
-		return
-	}
-
-	// TODO: Use requests to do actual updates with Admin API
-	tfNewModel, shouldReturn := mockedSDK(ctx, diags, plan.Timeouts)
-	// TODO: keep project_id and name from plan to avoid overwriting for move_state tests. We should probably do the same with the rest of attributes
-	tfNewModel.Name = plan.Name
-	tfNewModel.ProjectID = plan.ProjectID
-	if shouldReturn {
-		return
+	if cluster != nil {
```

Comment:
Updated in c8f4ac2954f5e9bf45b455e214c7630dcfe51b2a
WDYT?
Cluster == `nil` when there hasn't been any updates to cluster. Should I add a comment?

Reply:
Changed in 5d9c638645429e4ee35ec1822f201f1f63d41df5
I think so, to ensure any computed attribute has their latest value

Reply:
@EspenAlbert not sure it's needed, cc @AgustinBettati @maastha  in case you know.

looking at: https://developer.hashicorp.com/terraform/plugin/framework/resources/update
```
Only successfully modified parts of the resource should be return updated data in the state response.
```

I'm not sure we should update the state if there is not any change.

Reply:
I see we do read as the last step in Update in the current implementation, but not sure if it's just a precaution or it's really needed.

Reply:
cc @EspenAlbert

Reply:
We do this in this implementation. It is only when exclusively advanced_configuration has been updated we are doing an extra read